### PR TITLE
Add average_init_density to improve robustness of nerfacto training

### PR DESCRIPTION
**Pull request by jb-ye**
This has been my personal recipe for training nerfacto: the basic issue of current nerfacto implementation is the scene is always initialized with an average density at 1.0 (due to the exponential activation of MLP). This is not a robust choice in general especially when scene depth is much higher than 1.0. 

One famous example is the mipnerf360 `stump` dataset, when training with the recommended setting before this PR
```
ns-train nerfacto --data stump \
  --pipeline.model.appearance_embed_dim 1 \
  --pipeline.model.camera_optimizer.mode off
```
The PSNR on eval set is only 18. With this PR, one can choose to train with a multiplier on top of density output. My personal experience is to set it 0.01 so the scene is very transparent initially. 
```
ns-train nerfacto --data stump \
  --pipeline.model.appearance_embed_dim 1 \
  --pipeline.model.camera_optimizer.mode off \
  --pipeline.model.average_init_density 0.01
```
The PSNR on eval set will be improved from 18 to 25.3. Similar improvement can also be observed in some of nerfstudio dataset.

At last, this feature would not produce more quality in the final render, but will avoid some bad optimization outcomes due to improper initialization.

### CODE REVIEW

File: nerfstudio/fields/density_fields.py [hunk 1]
```diff
@@ -66,6 +67,7 @@ def __init__(
         self.register_buffer("max_res", torch.tensor(max_res))
         self.register_buffer("num_levels", torch.tensor(num_levels))
         self.register_buffer("log2_hashmap_size", torch.tensor(log2_hashmap_size))
+        self.register_buffer("average_init_density", torch.tensor(average_init_density))
```
Comment (jkulhanek) [inline nerfstudio/fields/density_fields.py:70]: Why is this a buffer here?

Comment (jb-ye) [reply nerfstudio/fields/density_fields.py:70]: This is my question as well, I saw other configurable parameters are set as buffer. I just tried to mimic the code style.

Anyone who can explain why `max_res`, `num_levels`, `log2_hashmap_size` are set as buffer?

Comment (jkulhanek) [reply nerfstudio/fields/density_fields.py:70]: Seems a bit wasteful to me.

Comment (jkulhanek) [reply nerfstudio/fields/density_fields.py:70]: Also it makes the new checkpoint not compatible with the previous one.

Comment (tancik) [reply nerfstudio/fields/density_fields.py:70]: It was originally changed here - https://github.com/nerfstudio-project/nerfstudio/pull/1486

Comment (jkulhanek) [reply nerfstudio/fields/density_fields.py:70]: In the case of aabb, it was a parameter before. It makes sense as it is a tensor. For scalars I think it’s better to keep them as floats?

Comment (brentyi) [reply nerfstudio/fields/density_fields.py:70]: Probably Volinga wanted it so they could read config values from the checkpoint's state_dict? But it also seems weird to me.

File: nerfstudio/configs/method_configs.py [hunk 2]
```diff
@@ -97,6 +97,7 @@
         ),
         model=NerfactoModelConfig(
             eval_num_rays_per_chunk=1 << 15,
+            average_init_density=0.01,
```
Comment (kerrj) [inline nerfstudio/configs/method_configs.py:100]: maybe add these to nerfacto-big and nerfacto-huge too?

Comment (kerrj) [reply nerfstudio/configs/method_configs.py:100]: oops I'm blind you already did

**Review by kerrj — APPROVED** [review]
sgtm!

**Pull request merged**